packages/amplify-codegen-e2e-core/src/categories/codegen.ts

```diff
@@ -31,7 +31,11 @@ export function generateStatementsAndTypes(cwd: string) : Promise<void> {
 // CLI workflow to add codegen to Amplify project
 export function addCodegen(cwd: string, settings: any = {}): Promise<void> {
   return new Promise((resolve, reject) => {
-    const chain = spawn(getCLIPath(), ['codegen', 'add'], { cwd, stripColors: true });
+    const cmdOptions = ['codegen', 'add'];
+    if (settings.withoutInit) {
```

Comment:
I suggest adding this command into a new method called `addCodegenWithApiId` instead of depending on the flag value as they are two use cases. Also could you confirm that running `amplify codegen add` without init is not allowed(I remember the previous command will give a msg like "please use the apiId instead"?)

Reply:
I couldn't find reference to non-amplify workflows in the latest CLI docs but looking at this legacy page, the only requirement to is to have a local introspection schema and with or without apiId should work. https://docs.amplify.aws/cli-legacy/graphql-transformer/codegen/#amplify-add-codegen